stonesoup/moveable/movable.py

```diff
@@ -0,0 +1,532 @@
+
+            rot = _get_rotation_matrix(vel)
+            return rot @ offset
+        else:
+            return offset
+
+    def range_and_angles_to_other(self, other: 'Movable') -> Tuple[float, float, float]:
+        """ Calculate the range, azimuth and elevation of a given Movable relative to current
+        Movable.
+
+        Calculates the relative vector between the two Movables, and then converts this
+        range, azimuth, elevation using :func:`.cart2sphere`
+
+        Parameters
+        ----------
+        other : :class:`~.Movable`
+            Another Movable. Only its position is relevant to this method.
+
+        Returns
+        -------
+        range, azimuth, elevation : :class:`float`, :class:`float`, :class:`float`
+            The range azimuth and elevation of the target from the radar
+
+        """
+        # state relative to radar (in cartesian space)
+        relative_vector = other.position - self.position
+        relative_vector = self._rotation_matrix @ relative_vector
+        # calculate target position in spherical coordinates
+        [range_, azimuth, elevation] = cart2sphere(*relative_vector)
+        return range_, azimuth, elevation
+
+    @property
+    def _rotation_matrix(self) -> np.ndarray:
+        """_rotation_matrix getter method
+
+        Calculates and returns the (3D) axis rotation matrix.
+
+        Returns
+        -------
+        : :class:`~numpy.ndarray` of shape (3, 3)
+            The model (3D) rotation matrix.
+        """
+        return build_rotation_matrix(self.orientation)
+
+
+class FixedMovable(Movable):
+    """Fixed platform base class
+
+        A platform represents a random object defined as a :class:`~.StateMutableSequence`
+        with fixed (but settable) position and orientation.
+
+        .. note:: Position and orientation are read/write properties in this class.
+        """
+    orientation: StateVector = Property(
+        default=None,
+        doc='A fixed orientation of the static platform. Defaults to the zero vector')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.orientation is None:
+            self.orientation = StateVector([0, 0, 0])
+
+    def _set_position(self, value: StateVector) -> None:
+        self.state_vector[self.position_mapping, :] = value
+
+    @property
+    def velocity(self) -> StateVector:
+        """Return the velocity of the platform.
+
+        For a fixed platform this is always a zero vector of length :attr:`ndim`.
+        """
+        return StateVector([0] * self.ndim)
+
+    @property
+    def is_moving(self) -> bool:
+        return False
+
+    def move(self, timestamp: datetime.datetime, **kwargs) -> None:
+        """For a fixed platform this method has no effect other than to update the timestamp."""
+        # TODO Is this a sensible implementation?
+        # Return without moving static platforms
+        self.state.timestamp = timestamp
```

Comment:
While the state vector remains the same, as the timestamp changes, should this 'new' state be appended to :attr:`states`?

Reply:
Well spotted. Fixed.